### PR TITLE
Add support for cnn 3d NDHWC

### LINKED ISSUE #9561: Follow-up: Creation of empty UTF-8 (String) arrays will fail
A follow-up to https://github.com/eclipse/deeplearning4j/issues/9540, as it looks like the described issue is present in 8 more of the various create methods in `Nd4j`.

Wouldn't it make sense to change the other methods as well, where `Nd4j.scalar([dataTypeAsGivenByVariableOrMethod], 0.0)` is returned in case `shape.length == 0`? I assume, these will lead to the same issue in different contexts, if the data structure is empty and the type is UTF-8.

#### Version Information

* Deeplearning4j/Nd4j version: 1.0.0-M1
* Platform information: Ubuntu 18.04.6 LTS
* CUDA version: 11.5
* NVIDIA driver version: 495.29.05

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Adds missing data format configurations for relevant cnn3d layers (pooling, conv, upsampling)

Fixes https://github.com/eclipse/deeplearning4j/issues/9561

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [X ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ X] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [X ] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

### CODE REVIEW

File: deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/layers/convolution/KerasConvolution3DTest.java [hunk 1]
```diff
@@ -145,7 +151,7 @@ private void buildConvolution3DLayer(KerasLayerConfiguration conf, Integer keras
         config.put(conf.getLAYER_FIELD_BORDER_MODE(), BORDER_MODE_VALID);
         layerConfig.put(conf.getLAYER_FIELD_CONFIG(), config);
         layerConfig.put(conf.getLAYER_FIELD_KERAS_VERSION(), kerasVersion);
-        ConvolutionLayer layer = new KerasConvolution3D(layerConfig).getConvolution3DLayer();
+        Convolution3D layer = (Convolution3D) new KerasConvolution3D(layerConfig).getConvolution3DLayer();
```
Comment (treo) [inline deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/layers/convolution/KerasConvolution3DTest.java]: Is that cast still necessary here?

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java [hunk 2]
```diff
@@ -3714,7 +3714,7 @@ public static INDArray create(byte[] data, long[] shape, long[]strides, char ord
     }
 
     /**
-     * See {@link #create(int[], long[], long[], char, DataType)}
+     * See {@link #, long[], long[], char, DataType)}
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: Did you accidentally delete part of the name here?

**Review by treo — CHANGES_REQUESTED** [review]

File: deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/layers/convolution/KerasConvolution3DTest.java (hunk 1 above)
Comment (agibsonccc) [reply deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/layers/convolution/KerasConvolution3DTest.java]: It was returning a ConvolutionLayer. Probably not now.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java (hunk 2 above)
Comment (agibsonccc) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java]: Indeed I did

**Review by treo — APPROVED** [review]

**Pull request merged**